pkg/odo/cli/preference/unset.go

```diff
@@ -0,0 +1,108 @@
+package preference
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/redhat-developer/odo/pkg/odo/cli/ui"
+
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/preference"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+)
+
+const unsetCommandName = "unset"
+
+var (
+	unsetLongDesc = ktemplates.LongDesc(`Unset an individual value in the Odo preference file.
+
+%[1]s
+%[2]s
+`)
+	unsetExample = ktemplates.Examples(`
+   # Unset a preference value in the global preference
+   %[1]s  %[2]s 
+   %[1]s  %[3]s 
+   %[1]s  %[4]s 
+	`)
+)
+
+// UnsetOptions encapsulates the options for the command
+type UnsetOptions struct {
+	paramName           string
+	preferenceForceFlag bool
+}
+
+// NewUnsetOptions creates a new UnsetOptions instance
+func NewUnsetOptions() *UnsetOptions {
+	return &UnsetOptions{}
+}
+
+// Complete completes UnsetOptions after they've been created
+func (o *UnsetOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.paramName = args[0]
+	return
+}
+
+// Validate validates the UnsetOptions based on completed values
+func (o *UnsetOptions) Validate() (err error) {
+	return
+}
+
+// Run contains the logic for the command
+func (o *UnsetOptions) Run() (err error) {
+
+	cfg, err := preference.New()
+
+	if err != nil {
+		return errors.Wrapf(err, "")
+	}
+
+	if value, ok := cfg.GetConfiguration(o.paramName); ok && (value != nil) {
```

Comment:
I would put this code in `Validate` and use the same logic as used in `Set` to make things simpler to understand (since the logic is fundamentally the same) and the logic here is more difficult to follow, in my opinion.

Reply:
used the same logic as set but still having the same query as https://github.com/redhat-developer/odo/pull/1362#discussion_r